elements/rh-secondary-nav/rh-secondary-nav-menu.ts

```diff
@@ -0,0 +1,163 @@
+import { classMap } from 'lit/directives/class-map.js';
+
+import { pfelement, observed } from '@patternfly/pfe-core/decorators.js';
+import { getRandomId } from '@patternfly/pfe-core/functions/random.js';
+
+import { tabletLandscapeBreakpoint } from '../../lib/tokens.js';
+import { MatchMediaController } from '../../lib/MatchMediaController.js';
+
+import { RhSecondaryNavDropdown } from './rh-secondary-nav-dropdown';
+
+import styles from './rh-secondary-nav-menu.css';
+
+/**
+ * @summary 'A pop up menu for secondary nav, available in full-width and fixed-with sizes'
+ *
+ * @slot base             - Slot to override all shadow dom contents for full-width or fixed-width menus
+ * @slot section          - Hidden if empty, section slot applies an auto column grid system for `<ul>` link lists
+ * @slot links            - Only for type="fixed-width", expects a `<ul>`
+ * @slot cta              - Menu level CTA, expects a `<pfe-cta>`
+ *
+ * @csspart full-width    - {HTMLElement} container - The wrapper for full-width menus
+ * @csspart fixed-width   - {HTMLElement} container - The wrapper for fixed-width menus
+ * @csspart sections      - {HTMLElement} container - The wrapper for menu sections
+ * @csspart cta           - {HTMLElement} container - The wrapper for cta
+ */
+@customElement('rh-secondary-nav-menu') @pfelement()
+export class RhSecondaryNavMenu extends LitElement {
+  static readonly styles = [styles];
+
+  // TODO: change to variant instead of type which is used by html element properties such as input.
+  @property({ reflect: true }) type: 'fixed-width' | 'full-width' = 'full-width';
+
+  @queryAssignedNodes('cta', true)
+  private _ctaNodes: NodeListOf<HTMLElement> | undefined;
+
+  @queryAssignedNodes('section', true)
+  private _sectionsNodes: NodeListOf<HTMLElement> | undefined;
+
+  @query('#sections') _sections: HTMLElement | undefined;
+
+  @query('#cta') _cta: HTMLElement | undefined;
+
+  @query('#container') _container: HTMLElement | undefined;
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.id ||= getRandomId('rh-secondary-nav-menu');
+  }
+
+  /**
+   * `isMobile` property is true when viewport `(min-width: ${tabletLandscapeBreakpoint})`.
+   * Property is observed for changes, and its value is updated using matchMediaControler
+   * when viewport changes at breakpoint or first load of the component.
+   */
+  @observed
+  @property({ reflect: false }) _isMobile = false;
+
+  /**
+   * Checks if passed in element is a RhSecondaryNavDropdown
+   * @param element:
+   * @returns {boolean}
+   */
+  static isDropdown(element: Element|null): element is RhSecondaryNavDropdown {
+    return element instanceof RhSecondaryNavDropdown;
+  }
+
+  /**
+   * Updates isMobile property with matchMedia when viewport changes
+   */
+  protected matchMedia = new MatchMediaController(this, `(min-width: ${tabletLandscapeBreakpoint})`, {
```

Comment:
can we use RHDSScreenSizeController here instead, so that we don't have to import tokens here? you also wouldn't have to set a state property either, just 
```ts
const { mobile } = this.#screenSize;
return html`<div class=${classMap({ mobile })}></div>`;
```

Reply:
That's muy caliente 🌶️ !  Yes definitely!